src/pybind/mgr/cephadm/upgrade.py

```diff
@@ -494,6 +494,35 @@ def upgrade_stop(self) -> str:
         self.mgr.event.set()
         return 'Stopped upgrade to %s' % target_image
 
+    def upgrade_service(self, service_type: str, service_image: str, image: str) -> List[str]:
+        key = f'mgr/cephadm/container_image_{service_image}'
+        try:
+            self.mgr.check_mon_command({
+                'prefix': 'config set',
+                'name': key,
+                'value': image,
+                'who': 'mgr',
+            })
+        except MonCommandFailed as e:
+            self.mgr.log.exception(f"Failed to set {service_image} option {key}")
+            raise OrchestratorError(f"Failed to set {service_image} option {key}: {e}")
+
+        out = []
+        # Redeploy all the service daemons of provided service type
+        for service_name, spec in self.mgr.spec_store.all_specs.items():
+            if service_type != spec.service_type:
+                continue
+            self.mgr.log.debug("Redeploying daemons of service {service_name}")
```

Comment:
I don't think we need this log line. `perform_service_action` already logs the service action it's performing